src/sources/kubernetes/file_source_builder.rs

```diff
@@ -0,0 +1,222 @@
+        // possible combination of uid, container_name, and namespace.
+        for uid in pod_uids {
+            for container_name in container_names {
+                for namespace in namespaces {
+                    self.file_config.include.push(
+                        (LOG_DIRECTORY.to_owned()
+                            + format!("{}_*_{}/{}/*.log", namespace, uid, container_name).as_str())
+                        .into(),
+                    );
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Configures exclude in FileConfig
+    fn file_source_exclude(&mut self) {
+        // True if there is no includes
+        let no_include = self.config.include_container_names.is_empty()
+            && self.config.include_namespaces.is_empty()
+            && self.config.include_pod_uids.is_empty();
+
+        // Default excludes
+        if no_include {
+            // Since there is no user intention in including specific namespace/pod/container,
+            // exclude kuberenetes and vector logs.
+
+            // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
+            // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+
+            // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+        }
+    }
+
+    /// Checks included uids for validity, and transforms them to numerical format.
+    fn numerical_pod_uids(&self) -> Result<Vec<String>, BuildError> {
+        let mut include_pod_uids = Vec::new();
+        for pod_uid in &self.config.include_pod_uids {
+            let mut pod_uid = pod_uid.clone();
+
+            // Check if pod_uid contains only UID valid characters
+            if !pod_uid.chars().all(|c| {
+                c.is_numeric() || ('A'..='F').contains(&c) || ('a'..='f').contains(&c) || c == '-'
+            }) {
+                error!(message = "Configuration 'include_pod_uids' contains not UID", uid = ?pod_uid);
+                return Err(BuildError::IllegalCharacterInUid {
+                    uid: pod_uid.clone(),
+                });
+            }
+
+            // Remove dashes from uid so to always have numerical format
+            // of uid.
+            pod_uid.retain(|c| c != '-');
+
+            if pod_uid.chars().count() > 32 {
+                return Err(BuildError::UidToLarge {
+                    uid: pod_uid.clone(),
+                });
+            }
+
+            include_pod_uids.push(pod_uid);
+        }
+
+        Ok(include_pod_uids)
+    }
+}
+
+/// Transforms uid to two UID forms defined by https://tools.ietf.org/html/rfc4122.
```

Comment:
I know we've been using `UID` but this rfc talks about `UUID`, what are the relationships here? In relation to kube, it uses uid but does it actually follow rfc4122?

Reply:
And another question, is can we use the `uuid` crate, I feel like some of this logic is probably duplicated here?

Reply:
Kubernetes is calling them UID, but they are also UUID by that standard. https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids mentions ISO standard, but rfc one is technically the same, but more accessible.   

We can't, at least not without some hackiness. The problem is that we are never actually handling UUIDs. The input ones can be shorter, so they aren't UUID, and later we pad them with non UUID characters and make them larger than a real UUID.  `uuid` crate is quite strict on this, and doesn't allow it, which is good.

Reply:
Cool if that is the case, I'd like to make sure we isolate the behavior and test it very well.

Reply:
It was simplified and refactored into `PartialUid`, so added tests should be enough.